src/main/java/by/exadel/internship/service/impl/SchedulingServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+package by.exadel.internship.service.impl;
+
+import by.exadel.internship.dto.InterviewDTO;
+
+import by.exadel.internship.dto.time_for_call.UserTimeSlotWithUserIdDTO;
+import by.exadel.internship.dto.UserDTO;
+import by.exadel.internship.dto.enums.FormStatus;
+import by.exadel.internship.dto.enums.UserRole;
+import by.exadel.internship.dto.form.FormFullDTO;
+import by.exadel.internship.exception_handing.InappropriateRoleException;
+import by.exadel.internship.service.*;
+import by.exadel.internship.util.MDCLog;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class SchedulingServiceImpl implements SchedulingService {
+    private final UserTimeSlotService userTimeSlotService;
+    private final FormService formService;
+    private final UserService userService;
+
+    private static final Map<FormStatus, UserRole> STATUS_USER_ROLE_MAP = new HashMap<>() {
+        {
+            put(FormStatus.REGISTERED, UserRole.ADMIN);
+            put(FormStatus.ADMIN_INTERVIEW_PASSED, UserRole.TECH_EXPERT);
+        }
+    };
+
+    private static final String SIMPLE_CLASS_NAME = SchedulingService.class.getSimpleName();
+
+
+    @Override
+    public List<UserDTO> getAdminTimeForForm(UUID formId) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to get free users time for Form");
+        FormFullDTO formFullDTO = formService.getById(formId);
+        return checkInterviewOver(formFullDTO);
+    }
+
+    private List<UserDTO> checkInterviewOver(FormFullDTO formFullDTO) {
+        FormStatus status = formFullDTO.getFormStatus();
+        if (status.equals(FormStatus.REGISTERED)) {
```

Comment:
Simplify it. STATUS_USER_ROLE_MAP.get(formStatus) save to local variable and wrap by Optional. than call orElseThrow. You don't need to repeat call `findAdminForInterview`

Reply:
I don't understand what needs to be changed. Because we don't repeat call findAdminForInterview. We call only once for form :)